### PR TITLE
Solved

### PR DESCRIPTION
**Pull request by creewick**

### CODE REVIEW

**Comment by manwar** [comment]
Thanks @creewick

File: challenge-030/creewick/ch-1.pl [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/perl
```
Comment (k-mx) [inline challenge-030/creewick/ch-1.pl]: `#!/usr/bin/env perl` is better. On some systems, interpreter may be situated in another place. And perlbrew won't work with such shebang

File: challenge-030/creewick/ch-1.pl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        : $year % 100 > 0
+            ? 1
+            : $year % 400 == 0;
+}
+
+while ($currentYear < 2100) {
+    my $isLeap = isLeapYear(++$currentYear);
+    $currentDayOfWeek = getNextWeekDay($currentDayOfWeek, $isLeap);
+
+    if ($currentDayOfWeek == $REQUESTED_DAY_OF_WEEK) {
```
Comment (k-mx) [inline challenge-030/creewick/ch-1.pl]: You can use shorter postfix form:
`say $currentYear if $currentDayOfWeek = $REQUESTED_DAY_OF_WEEK;`

File: challenge-030/creewick/ch-2.pl [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    } else {
+        for my $i (1..$REQUIRED_SUM) {
+            my @newSeries = @series;
+            push(@newSeries, $i);
+            checkAllSeries(@newSeries);
+        }
+    }
+}
+
+checkAllSeries(());
```
Comment (k-mx) [inline challenge-030/creewick/ch-2.pl]: () are useless, so:
`checkAllSeries(());` -> `checkAllSeries();`

File: challenge-030/creewick/ch-1.pl [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use v5.10;
+
+# And you can change these constants to get the solution for day other than Christmas, or for other day of week
+
+my $currentDayOfWeek = 3;
+my $currentYear = 2019;
+my $REQUESTED_DAY_OF_WEEK = 0;
```
Comment (k-mx) [inline challenge-030/creewick/ch-1.pl]:
```
use constant {
    REQUESTED_DAY_OF_WEEK => 0,
    ...
};
```

**Pull request merged**